log10/load.py

```diff
@@ -34,24 +34,56 @@
 elif target_service is None:
     target_service = "log10"  # default to log10
 
+errors = (
+    error.APIConnectionError,
+    error.APIError,
+    error.RateLimitError,
+    error.ServiceUnavailableError,
+    error.Timeout,
+)
 
-@backoff.on_exception(backoff.expo, (RateLimitError, APIConnectionError))
+
+@backoff.on_exception(backoff.expo, errors)
 def func_with_backoff(func, *args, **kwargs):
     return func(*args, **kwargs)
 
 
+# todo: should we do backoff as well?
```

Comment:
@kxtran this is the current implementation in log10/openai.py. Based on your comment above, would be better to retry `openai.Completion.create` instead of the whole text function, which will create new completion_id for every retry?
```
    @backoff.on_exception(backoff.expo, RETRY_ERROR_TYPES)
    def text(self, prompt: str, hparams: dict = None) -> TextCompletion:
        request = self.text_request(prompt, hparams)

        start_time = time.perf_counter()
        completion_id = self.log_start(request, Kind.text)

        completion = openai.Completion.create(**request)
        response = TextCompletion(text=completion.choices[0].text, response=completion)

        self.log_end(
            completion_id,
            completion,
            time.perf_counter() - start_time,
        )

        return response
```

Reply:
@kxtran does the pseudo-code below align with your comment?

```
# if fail to create completionId, then retry creating until succeed
# we could set the time limit, like max tries and max time
@backoff with retry
def log_start -> completionID

# retry for openai calls, openai doesn't handle retry
# other providers could be different, anthropic does retry two times. 
@backoff with retry
def openai_call -> completion

def text:
    log_start()
    openai_call()
```